radixdlt-engine/src/main/java/com/radixdlt/atom/Atom.java

```diff
@@ -77,13 +79,21 @@
 	private final Map<EUID, ECDSASignature> signatures = new HashMap<>();
 
 	public Atom() {
-		this(null);
+		this(ImmutableList.of(), Map.of(), null);
 	}
 
 	public Atom(@Nullable String message) {
 		this.message = message;
 	}
 
+	public Atom(ParticleGroup pg) {
+		this(List.of(pg), Map.of(), null);
+	}
+
+	public Atom(List<ParticleGroup> pgs) {
+		this(pgs, Map.of(), null);
+	}
+
 	public Atom(List<ParticleGroup> particleGroups, Map<EUID, ECDSASignature> signatures, @Nullable String message) {
 		Objects.requireNonNull(particleGroups, "particleGroups is required");
 		Objects.requireNonNull(signatures, "signatures is required");
```

Comment:
It's a good practice to avoid exceptions in the constructor. For this purpose most often used approach is the `single constructor + several factory methods` instead of several constructors with different set of parameters. I think we should continue use of factory methods as it was before.

Reply:
yes agreed, this will be fixed in upcoming PR regarding atom refactor